avocado/core/job.py

```diff
@@ -224,6 +201,37 @@ def __enter__(self):
     def __exit__(self, _exc_type, _exc_value, _traceback):
         self.cleanup()
 
+    @property
+    def test_parameters(self):
+        """Placeholder for test parameters.
+
+        This is related to --test-parameters command line option. They're kept
+        in the job because they will be prepared only once, since they are read
+        only and will be shared across all tests of a job.
+        """
+        if self._test_parameters is not None:
+            return self._test_parameters
+
+        self._test_parameters = {name: value for name, value
+                                 in self.config.get('run.test_parameters', [])}
+        return self._test_parameters
+
+    @property
+    def timeout(self):
+        if self._timeout is not None:
+            return self._timeout
+
+        self._timeout = self.config.get('job.run.timeout')
+        return self._timeout
```

Comment:
While reviewing the code again I saw a possibility for another refactor here:

```
    def timeout(self):
        if self._timeout is None:
            self._timeout = self.config.get('job.run.timeout')
        return self._timeout
```

What do you think? Sorry for the latest note on it.

Reply:
@willianrampazzo I liked it! Thanks. I did a force-push here fixed this and the other properties too.